django_tutorial/first_project/web_app/static_src/js/redux/color.js

```diff
@@ -0,0 +1,59 @@
+import {getRegions} from './regions'
+import {Quantiler} from '../utilities'
+
+/*
+ * Note that the color controller uses a 'slice' of the global state, i.e. the state
+ * we deal with here is stored in the 'color' field of the global state, or in other
+ * words, stateFromThisFile === globalState.color.
+ */
+
+
+ ////////////////// Initial State //////////////////////
+
+
+const initialState = {
+  colorField: 'sba_per_small_bus',
+  colorQuantiler: undefined,
+  numQuantiles: 10
+}
+
+
+
+////////////////// State Selectors //////////////////////
+
+
+export const getColorField         = (state) => state.colorField
+export const getNumColorQuantiles  = (state) => state.numQuantiles
+export const getColorQuantiler     = (state) => state.colorQuantiler
+
+
+////////////////// Action Creators //////////////////////
+
+export function setColorField(colorField) {
+  return {type: 'SET_COLOR_FIELD', colorField}
+}
+
+
+////////////////// Reducers //////////////////////
+
+/**
+ * Note that unlike a normal reducer, this reducer also accepts an optional regionState
+ * param since the color quantiler state depends on the region state
+ */
+export default function colorReducer(state=initialState, action={}, regionState) {
+  const {type, colorField=getColorField(state)} = action
```

Comment:
What exactly does the assignment of colorField do within the destructuring assignment?

`colorField=getColorField(state)`

Reply:
It provides a default value for colorField in case the action doesn't provide one.  In this case, the SET_DISTRICT_REGION_TYPE_AND_REGIONS action doesn't provide a colorField, so this is a shorthand way of leaving the colorField unchanged in the state.